src/System/IO/UTF8.hs

```diff
@@ -4,11 +4,21 @@ import Prelude.Compat
 
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.UTF8 as UTF8
+import           Data.Text (Text)
+import qualified Data.Text.Encoding as TE
+
+readUTF8FileT :: FilePath -> IO Text
+readUTF8FileT inFile =
+  fmap TE.decodeUtf8 (BS.readFile inFile)
```

Comment:
why not use https://hackage.haskell.org/package/text-1.2.2.1/docs/Data-Text-IO.html instead ?